lncrawl/sources/lightnovelpub.py

```diff
@@ -113,10 +113,7 @@ def download_chapter_body(self, chapter):
         '''Download body of a single chapter and return as clean html format.'''
         logger.info('Downloading %s', chapter['url'])
         soup = self.get_soup(chapter['url'])
-        body = soup.select_one('.chapter-content')
-        for ads in body.select('div[class*="ad"], script, ins'):
-            ads.decompose()
```

Comment:
The purpose of this loop was to remove ads in body. If you remove this, there might be text from ads in the content.

Reply:
Have you checked that removing this will not show any ads?

Reply:
This worked correctly, but turns out not all the novels use `<p>` tag, so some ended with empty chapters. I reverted this change and instead remove ads in this loop.